math/mathcore/src/FitUtil.cxx

```diff
@@ -591,19 +591,20 @@ double FitUtil::EvaluateChi2Residual(const IModelFunction & func, const BinData
 
    // expected errors
    if (useExpErrors) {
-      // we need first to check if a weight factor needs to be applied
-      // weight = sumw2/sumw = error**2/content
-      // NOTE: assume histogram is not weighted
-      // don't know how to do with bins with weight = 0
-      //double invWeight = y * invError * invError;
-      // if (invError == 0) invWeight = (data.SumOfError2() > 0) ? data.SumOfContent()/ data.SumOfError2() : 1.0;
+      // check if a weight factor needs to be applied
+      // for bins with y = 0 use as weight the global of the full histogram
```

Comment:
"for bins with y = 0 use as weight the global of the full histogram"

Is there literature somewhere that explains why this is the good solution? Maybe there should be a link to a paper?